references/classification/fastai/train.py

```diff
@@ -37,61 +75,94 @@ def main(args):
-
         labels.append(sample['target'])
         is_valid.append(True)
 
     df = pd.DataFrame.from_dict(dict(name=fnames, label=labels, is_valid=is_valid))
 
-    il = vision.ImageList.from_df(df, path=args.data_path).split_from_df('is_valid').label_from_df(cols='label')
+    # Split train and valid sets
+    il = vision.ImageList.from_df(df, path=args.data_path).split_from_df('is_valid')
+    # Encode labels
+    il = il.label_from_df(cols='label', label_cls=FloatList if args.binary else CategoryList)
+    # Set transformations
     il = il.transform(vision.get_transforms(), size=args.resize)
+    # Create the Databunch
     data = il.databunch(bs=args.batch_size, num_workers=args.workers).normalize(vision.imagenet_stats)
+    # Metric
+    metric = partial(vision.accuracy_thresh, thresh=0.5) if args.binary else vision.error_rate
+    # Create model
+    model = models.__dict__[args.model](imagenet_pretrained=args.pretrained,
+                                        num_classes=data.c, lin_features=args.lin_feats,
+                                        concat_pool=args.concat_pool, bn_final=args.bn_final,
+                                        dropout_prob=args.dropout_prob)
+    # Create learner
+    learner = Learner(data, model,
+                      wd=args.weight_decay,
+                      loss_func=CustomBCELogitsLoss() if args.binary else nn.CrossEntropyLoss(),
+                      metrics=metric)
+
+    # Form layer group for optimization
+    meta = model_meta.get(args.model, _default_meta)
+    learner.split(meta['split'])
+    # Freeze model's head
+    if args.pretrained:
+        learner.freeze()
 
-    learner = vision.cnn_learner(data, vision.models.__dict__[args.model],
-                                 pretrained=args.pretrained,
-                                 wd=args.weight_decay,
-                                 ps=args.dropout_prob,
-                                 concat_pool=args.concat_pool,
-                                 metrics=vision.error_rate)
     if args.resume:
         learner.load(args.resume)
     if args.unfreeze:
         learner.unfreeze()
 
     learner.fit_one_cycle(args.epochs, max_lr=slice(None, args.lr, None),
-                          div_factor=args.div_factor)
+                          div_factor=args.div_factor, final_div=args.final_div_factor)
 
     learner.save(args.checkpoint)
 
 
 if __name__ == "__main__":
     import argparse
     parser = argparse.ArgumentParser(description='PyroNear Classification Training with Fastai')
+    # Input / Output
     parser.add_argument('--data-path', default='./data', help='dataset')
+    parser.add_argument('--checkpoint', default='checkpoint', type=str, help='name of output file')
+    parser.add_argument('--resume', default=None, help='checkpoint name to resume from (default: None)')
+    # Architecture
     parser.add_argument('--model', default='resnet18', type=str, help='model')
```

Comment:
Since now there are many possibilities, you could add the default to the helper (explicitly or using ArgumentDefaultsHelpFormatter for all arguments)